python/sglang/srt/layers/moe/ep_moe/token_dispatcher.py

```diff
@@ -374,16 +377,22 @@ def combine(
             hidden_states, event, hook = self.combine_low_latency(
                 hidden_states, self.topk_idx, self.topk_weights, self.handle
             )
+
+        if self.async_finish:
+            event.current_stream_wait()
+
         self.handle = None
         return hidden_states.view(self.hidden_shape)
 
-    def combine_normal(self, x: torch.Tensor, handle: Tuple, previous_event=None):
+    def combine_normal(self, x: torch.Tensor, handle: Tuple):
+        previous_event = Buffer.capture() if self.async_finish else None
+
         combined_x, _, event = self.buffer_normal.combine(
             x,
             handle,
-            async_finish=False,
+            async_finish=self.async_finish,
             previous_event=previous_event,
-            allocate_on_comm_stream=False,
+            allocate_on_comm_stream=(previous_event is not None) and self.async_finish,
```

Comment:
Only the first condition is useful. If `previous_event` is not None, `self.async_finish` is always true.

Reply:
Yes, I originally did that because DeepEP has assertion like: https://github.com/deepseek-ai/DeepEP/blob/c4b8ffc37ca8c9a5da72d62930158dd51e9d0850/csrc/deep_ep.cpp#L229 so I want to make it exactly the same condition. But I am OK for both changing it or not changing it.